### ISSUE TITLE
BlockSeq indents issue

### ISSUE THREAD

**Issue by dannypike** [jbeder/yaml-cpp#631]
I'm new to this game, so apologies if this is wrong, but it seems to me that yaml-cpp is adding too many indents on block sequences. This is the "recommended" format for a typical block sequence:

```
map1:
- map2:
    value2_1: 0
    value2_2: 1
- map3:
    value3_1: 0
    value3_2: 1
```

whereas yaml-cpp indents the sequence unnecessarily:

```
map1:
  - map2:
      value2_1: 0
      value2_2: 1
 - map3:
      value3_1: 0
      value3_2: 1
```

I realise it's a small problem, but the size can add up when there are hundreds of thousands of lines.

I think that the problem lies in emitter.cpp at line 354 in 774f25800e6f19f4b927023c85d1389af322da5e:

```
void Emitter::BlockSeqPrepareNode(EmitterNodeType::value child) {
  const std::size_t curIndent = m_pState->CurIndent();
  const std::size_t nextIndent = curIndent + m_pState->CurGroupIndent();

  if (child == EmitterNodeType::NoType)
    return;

  if (!m_pState->HasBegunContent()) {
    if (m_pState->CurGroupChildCount() > 0 || m_stream.comment()) {
      m_stream << "\n";
    }
    m_stream << IndentTo(curIndent);    <=== should this be something like max(0, curIndent - 1)?
    m_stream << "-";
  }
```

but I can't build yaml-cpp under Windows/VS2017 in order to test it and submit a PR (I get a lot of errors about unknown overrides). Can you fix it and update the vcpkg for me?

Many thanks,

Dan

**Comment by jbeder** [comment]
If you can't build under Windows/VS2017, can you file an issue for that?

As for the main point of this issue: the format you suggest is "recommend" is an alternative version, and I'm not sure if one is recommended over the other. The general question is, if you have a sequence inside a map, in flow style, do you want to indent. I personally prefer indenting, to be honest.

**Comment by dannypike** [comment]
My original statement that about the "recommended" format may well be wrong; as I said this YAML stuff is new to me and I was reiterating a comment that I had received from others. I'm sorry if that caused any offense.

As it turns out, I was unable to use your library for serializing and deserializing the particular YAML files that I had to deal with. I am sure that yaml-cpp is a full and accurate implementation of the standard. but it turned out to be too powerful for what I needed.

The YAML files that I was dealing with last month were very simple but rather large (several hundred thousand lines each). They didn't use any of the clever features of YAML, but the indentation was quite deep - there were tens of thousands of maps in each file, each with sequences that themselves had maps etc. The extra indents really added up, nearly doubling the size of the file in some cases!

So, in the end, I chose to write my own simplistic functions. However, i do thank you for yaml-cpp; I am sure that I will use it at some point.
